aws/aws.go

```diff
@@ -227,10 +245,25 @@ func GetAllResources(targetRegions []string, excludeAfter time.Time, resourceTyp
 
 		logging.Logger.Infof("Checking region [%d/%d]: %s", count, totalRegions, region)
 
-		session, err := session.NewSession(&awsgo.Config{
-			Region: awsgo.String(region),
-		},
-		)
+		var cloudNukeSession *session.Session
+		var err error
+
+		// If no aws.Config was passed in externally, generate a new session using the default credentials
+		if externalcreds.Get() == nil {
+			cloudNukeSession, err = session.NewSession(&awsgo.Config{
+				Region: awsgo.String(region),
+			})
+		} else {
+
+			externalConfig := externalcreds.Get()
+			// If a pointer to an aws.Config was passed by the caller, use its credentials
+			newConfig := &awsgo.Config{}
+			newConfig.WithRegion(region)
+			newConfig.WithCredentials(externalConfig.Credentials)
+
+			cloudNukeSession, err = session.NewSession(newConfig)
+		}
```

Comment:
Shouldn't this just use `newSession(region)`?

Reply:
Yup! Fixed in 23111ea